store/sqlstore/upgrade.go

```diff
@@ -915,3 +917,15 @@ func precheckMigrationToVersion528(sqlStore SqlStore) error {
 
 	return nil
 }
+
+func upgradeDatabaseToVersion529(sqlStore SqlStore) {
+	// if shouldPerformUpgrade(sqlStore, VERSION_5_28_0, VERSION_5_29_0) {
+
+	sqlStore.AlterColumnTypeIfExists("SidebarCategories", "Id", "VARCHAR(128)", "VARCHAR(128)")
+	sqlStore.AlterColumnDefaultIfExists("SidebarCategories", "Id", model.NewString(""), nil)
```

Comment:
Why is the default different between Postgres and MySQL?

Reply:
I'm honestly not sure why the schema is different with and without the upgrade. If I use a fresh database, I get
```
mysql> show columns in SidebarCategories;
+-------------+--------------+------+-----+---------+-------+
| Field       | Type         | Null | Key | Default | Extra |
+-------------+--------------+------+-----+---------+-------+
| Id          | varchar(128) | NO   | PRI | NULL    |       |
```
and if I run an upgrade without that line, I still get
```
mysql> show columns in SidebarCategories;
+-------------+--------------+------+-----+---------+-------+
| Field       | Type         | Null | Key | Default | Extra |
+-------------+--------------+------+-----+---------+-------+
| Id          | varchar(128) | NO   | PRI | NULL    |       |
```
but for some reason, without that line, the `test-schema` job failed for MySQL with 
```
Schema mismatch
547c547
<   `Id` varchar(128) NOT NULL DEFAULT '',
---
>   `Id` varchar(128) NOT NULL,
562c562
<   `CategoryId` varchar(128) NOT NULL DEFAULT '',
---
>   `CategoryId` varchar(128) NOT NULL,

Exited with code exit status 1
```

Since `AlterColumnDefaultIfExists` removes any default value when passed the empty string, it seems like altering the column type might be setting a default value on MySQL somehow

Reply:
Actually, I was trying to test again locally, and it seems like `make test-db-migration` doesn't work since it's not even giving me an error if I remove the migration entirely.

Reply:
I still haven't been able to come up with a better explanation about why the default seems to sometimes be set, but this at least ensures the column always correctly has no default even if this query might be redundant.

Reply:
Should we also make the postgres default to be empty string in that case?

Reply:
Both DBs are actually going to have no default after this. Passing the empty string to `AlterColumnDefaultIfExists` makes it clear the default ([link](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/supplier.go#L783)).

Reply:
Ugh sorry, I read it wrong :facepalm:

Reply:
No worries. It's kind of confusing anyway since it prevents us from actually making a column have the empty string as default. Now that I think of it, that special case might have actually caused me problems in the past.